lib/graphql/static_validation/literal_validator.rb

```diff
@@ -6,10 +6,10 @@ def validate(ast_value, type)
     elsif type.kind.list? && ast_value.is_a?(Array)
       item_type = type.of_type
       ast_value.all? { |val| validate(val, item_type) }
-    elsif type.kind.scalar?
-      !type.coerce_input(ast_value).nil?
+    elsif type.kind.scalar? && !ast_value.is_a?(GraphQL::Language::Nodes::AbstractNode) && !ast_value.is_a?(Array)
```

Comment:
What is the case here, where the type is scalar but the value is not a string, number or boolean?